src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -132,7 +132,32 @@ internal virtual string GetAssemblyFileVersion()
             {
                 var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
                 var filePath = Path.Combine(_clientDirectory, name);
-                return FileVersionInfo.GetVersionInfo(filePath).FileVersion;
+                var fileVersionInfo = FileVersionInfo.GetVersionInfo(filePath);
+
+                return $"{fileVersionInfo.FileVersion} ({ExtractShortSHA(fileVersionInfo.ProductVersion)})";
+            }
+
+            return "";
+        }
+
+        internal static string ExtractShortSHA(string productVersion)
+        {
+            if (productVersion != null)
+            {
+                string marker = "Commit Hash: "; // format used from AssemblyVersionAttribute
```

Comment:
We can very easily provide another attribute that has just the hash directly rather than having to parse this back out. Is there a reason we can't do that?